test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1178,7 +1178,11 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
           .to.be.instanceOf(Error)
           .to.have.property('name', 'MongoServerSelectionError');
 
-        expect(insertError).to.match(/connect ECONNREFUSED 127.0.0.1:27021/);
+        expect(insertError, 'Error must contain ECONNREFUSED').to.satisfy(
```

Comment:
Rather than hard-coding logic for different Node versions, why not inspect the error and search for `ECONNREFUSED`?
```suggestion
        expect(inspec(insertError), 'Error must contain ECONNREFUSED').to.match(/.*ECONNREFUSED.*/)
```

Reply:
I would want to know if there was yet another shape that the error took, the information went missing from the message in node 20, we will return it to the message in some way in: https://jira.mongodb.org/browse/NODE-5296, added todos and updated the ticket

Reply:
From our public docs about our errors:

> It is our recommendation to use instanceof checks on errors and to avoid relying on parsing error.message and error.name strings in your code. We guarantee instanceof checks will pass according to semver guidelines, but errors may be sub-classed or their messages may change at any time, even patch releases, as we see fit to increase the helpfulness of the errors.

We don't guarantee error messages.  The error information is still present, the error is just structured differently.  I don't think preserving the exact error info is important and so checking that the error itself is the correct error type is sufficient.

Reply:
Oh I was not thinking about public API / semver, just usefulness